tools/src/wyvern/tools/parsing/coreparser/WyvernParser.jj

```diff
@@ -459,12 +459,27 @@ Type Type() :
 		exp = build.invocation(exp, id.image, null, loc(id));
 	}
   	id = id2;
-  })* {
-	if (exp == null)
-		return build.nominalType(id.image, loc(id));
-  	else
-  		return build.qualifiedType(exp, id2.image);
-  }
+ })* (<TARROW> arrowResult = Type() )? {
+    if (arrowResult == null) {
```

Comment:
Good idea! Some of the abstractions were lost to me when I wrote that.
Reduces the cyclomatic complexity quiet a bit.